core/autoscaling.go

```diff
@@ -526,12 +566,29 @@ func (a *autoScalingGroup) launchCheapestSpotInstance(
 	}
 	logger.Println("Found on-demand instance", *baseInstance.InstanceId)
 
-	newInstanceType, err := baseInstance.getCheapestCompatibleSpotInstanceType()
-
+	newInstanceType, err = a.getInstanceTypeByTagInASG()
 	if err != nil {
-		logger.Println("No cheaper compatible instance type was found, "+
-			"nothing to do here...", err)
-		return errors.New("no cheaper spot instance found")
+		logger.Println("Errors while trying to fetch tags for instance type...")
+		return errors.New("Errors while trying to fetch tags")
+	}
+
+	// Check option to keep instance type
+	// If we keep the instance type we don't need to calculate the
+	// compatible instance type.
+	// Also ignore if it's already specified in the instance-type tag
+	if newInstanceType == "" {
+		if a.region.conf.KeepInstanceType == "current" {
+			newInstanceType = *baseInstance.InstanceType
+		} else if a.region.conf.KeepInstanceType != "" {
+			newInstanceType = a.region.conf.KeepInstanceType
```

Comment:
Exactly, I would prefer to implement that from the start, what you need is just a particular case of it and should need minor changes on top of what you already have done.

But if you're only willing to scratch your own itch, I understand and that's okay, we can enhance it to do  that later. It's already great that you are laying down a lot of the foundations for that extension and I can't thank you enough for this :)